modules/peridynamics/src/auxkernels/BondStatusPD.C

```diff
@@ -0,0 +1,93 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BondStatusPD.h"
+#include "MeshBasePD.h"
+
+registerMooseObject("PeridynamicsApp", BondStatusPD);
+
+template <>
+InputParameters
+validParams<BondStatusPD>()
+{
+  InputParameters params = validParams<AuxKernelBasePD>();
+  params.addClassDescription("Class for updating the bond status based on different failure "
+                             "criteria: critical stretch and "
+                             "maximum principal stress");
+  MooseEnum FailureCriteriaType("Critical_Stretch Maximum_Tensile_Stress", "Critical_Stretch");
```

Comment:
@hchen139 Says "Critical_Stretch Maximum_Tensile_Stress" as the first option ?

Reply:
Oh right, my bad I misread it

Since you mentioned that, maybe that could be another failure criteria too, based on the the tensile stress in the bond direction

Reply:
I have implemented the maximum tensile stress failure criterion before. Somehow, I decided to remove it from the current PR. If we need this failure criterion, we can definitely implement it in the future.